### PR TITLE
Fixes detected while preparing the run-through

### PR DESCRIPTION
**Pull request by mkuehbach**

### CODE REVIEW

**Comment by mkuehbach** [comment]
Run-through means we are testing the parser over a corpus of real datasets from the scientific community.

File: pynxtools_apm/utils/oasis_eln_reader.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ class NxApmNomadOasisElnSchemaParser:
     def __init__(self, file_path: str, entry_id: int, verbose: bool = False):
         print(f"Extracting data from ELN file: {file_path}")
         if (
-            file_path.rsplit("/", 1)[-1].startswith("eln_data")
-            or file_path.startswith("eln_data")
+            file_path.rsplit("/", 1)[-1].endswith("eln_data.yaml")
+            or file_path.rsplit("/", 1)[-1].endswith("eln_data.yml")
```
Comment (domna) [inline pynxtools_apm/utils/oasis_eln_reader.py:63]: Consider using `pathlib.Path(file_path).name`, which is a bit more expressive.
I think you can also allow any filename here and just check for the file ending (`os.path.splitext(file_path)[1] in (".yaml", ".yml")`)

File: pynxtools_apm/utils/oasis_config_reader.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def __init__(self, file_path: str, entry_id: int, verbose: bool = False):
         )
         if (
             file_path.rsplit("/", 1)[-1].endswith(".oasis.specific.yaml")
-            or file_path.endswith(".oasis.specific.yml")
+            or file_path.rsplit("/", 1)[-1].endswith(".oasis.specific.yml")
```
Comment (domna) [inline pynxtools_apm/utils/oasis_config_reader.py:40]: Maybe also `Path(file_path).name.endswith(".oasis.specific.yml")`

File: pynxtools_apm/utils/load_ranging.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def extract_data_from_rng_file(file_path: str, template: dict, entry_id: int) ->
 def extract_data_from_rrng_file(file_path: str, template: dict, entry_id) -> dict:
     """Add those required information which an RRNG file has."""
     print(f"Extracting data from RRNG file: {file_path}")
-    rangefile = ReadRrngFileFormat(file_path)
+    rangefile = ReadRrngFileFormat(file_path, unique=False)
```
Comment (domna) [inline pynxtools_apm/utils/load_ranging.py:236]: Not really necessary, because `unique` is the default. But fine if you want to have it for explicitness.

**Review by domna — APPROVED** [review]
Generally fine, I left a few suggestions for readability.

File: pynxtools_apm/utils/load_ranging.py (hunk 3 above)
Comment (mkuehbach) [reply pynxtools_apm/utils/load_ranging.py:236]: Exactly

File: pynxtools_apm/utils/oasis_eln_reader.py (hunk 1 above)
Comment (mkuehbach) [reply pynxtools_apm/utils/oasis_eln_reader.py:63]: implemented, no sticking with eln_data convention, indeed in the future should also use e.g anyname.eln and using the ELN file format

File: pynxtools_apm/utils/oasis_config_reader.py (hunk 2 above)
Comment (mkuehbach) [reply pynxtools_apm/utils/oasis_config_reader.py:40]: implemented

**Pull request merged**